src/amplitude-client.js

```diff
@@ -365,6 +365,23 @@ AmplitudeClient.prototype._trackParamsAndReferrer = function _trackParamsAndRefe
   }
 };
 
+/**
+ * Collect all utm, referral, and gclid data and combine into a single object
+ * @private
+ */
+
+AmplitudeClient.prototype._collectParamsAndReferrer = function _collectParamsAndReferrer() {
+  const utmProperties = this._getUtmProperties();
```

Comment:
you should consider only collecting these if the option to get them is turned on - to mirror `_trackParamsAndReferrer`

Reply:
naming nit: `_collectParamsAndReferrer` -> `_getParamsAndReferrer` or all of the below `get` to `collect`

Reply:
great call on both.  I changed the naming to `_getParamsAndReferrer` and I updated it to only fetch each piece of data if the functionality has been enabled in options